sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -144,39 +141,26 @@ public static BinaryData fromStream(InputStream inputStream) {
             .flatMap(bytes -> Mono.just(fromBytes(bytes)));
     }
 
-    /**
-     * Create {@link BinaryData} instance with given data and character set.
-     *
-     * <p><strong>Create an instance from String</strong></p>
-     * {@codesnippet com.azure.core.experimental.util.BinaryDocument.from#String}
-     *
-     * @param data to use.
-     * @param charSet to use.
-     * @throws NullPointerException if {@code inputStream} is null.
-     * @return {@link BinaryData} representing the binary data.
-     */
-    public static BinaryData fromString(String data, Charset charSet) {
-        Objects.requireNonNull(data, "'data' cannot be null.");
-
-        return new BinaryData(data.getBytes(charSet));
-    }
-
     /**
      * Create {@link BinaryData} instance with given data. The {@link String} is converted into bytes  using
-     * {@link StandardCharsets#UTF_8} character set.
+     * {@link StandardCharsets#UTF_8} character set. If {@code null} data is provided , it will be converted into
+     * empty byte array.
      *
      * @param data to use.
-     * @throws NullPointerException if {@code inputStream} is null.
      * @return {@link BinaryData} representing binary data.
      */
     public static BinaryData fromString(String data) {
-        Objects.requireNonNull(data, "'data' cannot be null.");
+        if (Objects.isNull(data) || data.length() == 0) {
+            return new BinaryData(EMPTY_DATA);
```

Comment:
new up a static final instance of this BinaryData so that you don't need to do it every time, and just reuse that.